data_steward/cdr_cleaner/cleaning_rules/README.md

```diff
@@ -0,0 +1,89 @@
+# Cleaning Rules
+All cleaning rules should be implemented as classes derived from `BaseCleaningRule`. The main objective of a cleaning rule is to generate queries returned by `get_query_specs()`. To do this a cleaning rule should use parameters `project_id`, `dataset_id` provided at runtime (e.g. when running `clean_cdr.py`).
+
+Some initial setup such as loading lookup tables may be needed for a cleaning rule to generate working queries; such setup should be done in `setup_rule()`. Cleaning rules should store lookups, intermediate data, and backup data in the location indicated by the `sandbox_dataset_id` parameter.
+
+## Implementation guidance
+ 
+- Cleaning rules should ideally be [idempotent](https://en.wikipedia.org/wiki/Idempotence)
+- Cleaning rules execute independently from one another, but may have implicit data interdependencies- be mindful of the order in which they run in the cleaning process.
+- If you are authoring a cleaning rule which requires some parameter other than the [universal parameters](#universal-parameters), check for a suitable [variable parameter](#variable-parameters) before creating a custom one specific for your cleaning rule and ensure that is documented in the same way as other parameters.
+
+# Cleaning rule configuration
+Cleaning rules depend on configuration parameters which must be supplied at runtime when cleaning a dataset. Some parameters are needed by all cleaning rules whereas others may only pertain to a few. These dependencies become difficult to manage without some overarching design. This section describes the design objective and summarizes existing configuration dependencies.
+
+## Parameters correctly supplied
+> Note: for these examples assume that cleaning fitbit only requires the additional parameters `mapping_dataset_id` and `pid_rid_map_tablename`
+
+The primary use case is clean_cdr is supplied all required parameters for all cleaning rules associated with the data stage. These parameters are successfully passed along to any cleaning rules that reference the parameter.
+```
+clean_cdr.py --data_stage FITBIT
+             --project_id my_project
+             --dataset_id fitbit
+             --sandbox_dataset_id fitbit_sandbox
+             --mapping_dataset_id mapping
+             --pid_rid_map_tablename pid_rid_map
+```
+
+## Parameters missing
+Another scenario is when `clean_cdr` is **NOT** supplied all the parameters needed by the cleaning rules associated with a run. The missing configuration is detected by `clean_cdr` prior to executing any of the cleaning rules. 
+```
+clean_cdr.py --data_stage FITBIT
+             --project_id my_project
+             --dataset_id fitbit
+             --sandbox_dataset_id fitbit_sandbox
+
+MissingParameterError: the parameter 'mapping_dataset_id' required by generate_ext_tables, RemoveFitbitDataIfMaxAgeExceeded, PIDtoRID FitbitDateShift was not supplied. No cleaning rules have been applied.
```

Comment:
It currently loops through the cleaning rules for parameter validation in `clean_cdr.py` and raises an error on the first cleaning rule that is missing required parameters.

Reply:
This might be nice, but this would be an enhancement and not base line functionality.  I think we are getting down to making sure the baseline functionality exists.